certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/validation/model/ElementValidationDateRange.java

```diff
@@ -0,0 +1,140 @@
+package se.inera.intyg.certificateservice.domain.validation.model;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Value;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementData;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValue;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDateRange;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementId;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.FieldId;
+
+@Value
+@Getter(AccessLevel.NONE)
+@Builder
+public class ElementValidationDateRange implements ElementValidation {
+
+  private static final String TO_SUFFIX = ".to";
+  private static final String FROM_SUFFIX = ".from";
+  private static final String RANGE_SUFFIX = ".range";
+  boolean mandatory;
+
+  @Override
+  public List<ValidationError> validate(ElementData data,
+      Optional<ElementId> categoryId, List<ElementData> dataList) {
+    if (data == null) {
+      throw new IllegalArgumentException("Element data is null");
+    }
+
+    final var dateRange = getValue(data.value());
+
+    final var rangeValidationErrors = getRangeValidationErrors(data, categoryId, dateRange);
```

Comment:
Consider moving the ```rangeValidationErrors``` so that they are done after we made the mandatory check. I think that might help to simplify the logic added in ```getRangeValidationErrors``` and we can remove: 

```java
final var incompleteErrors = getIncompleteDateRangeErrors(data, categoryId, dateRange);
```

and instead just use

```java
private List<ValidationError> getToBeforeFromDateRangeErrors(ElementData data,
      Optional<ElementId> categoryId,
      ElementValueDateRange dateRange) {
    return isToBeforeFrom(dateRange) ?
        List.of(errorMessage(
                data, getFieldId(dateRange.id(), RANGE_SUFFIX), categoryId,
                ErrorMessageFactory.endDateAfterStartDate()
            )
        ) : Collections.emptyList();
  }
```

Reply:
We can't replace incompleteErrors with mandatory since the field ids are different.

Reply:
I see.

Would it be a good idea so separate the logic in ```getIncompleteDateRangeErrors``` and validate the ```getIncompleteDateRangeErrors``` in a separate method?

I just found ```getRangeValidationErrors``` alittle complex to grasp and I think that would help and make the code easier to follow and understand.

Reply:
The logic is already separated and validated in separate methods